src/main/java/org/skriptlang/skript/lang/converter/Converters.java

```diff
@@ -304,8 +304,14 @@ private static <F, T extends ParentType, SubType extends F, ParentType> Converte
 
 		// Attempt to find converters that have either 'from' OR 'to' not exactly matching
 		for (ConverterInfo<?, ?> unknownInfo : CONVERTERS) {
+			int flags = unknownInfo.getFlags();
 			if (unknownInfo.getFrom().isAssignableFrom(fromType) && unknownInfo.getTo().isAssignableFrom(toType)) {
 				ConverterInfo<F, ParentType> info = (ConverterInfo<F, ParentType>) unknownInfo;
+				if ((flags & Converter.ALLOW_UNSAFE_CASTS) == 0) {
+					if ((flags & Converter.NO_RIGHT_CHAINING) == Converter.NO_RIGHT_CHAINING
```

Comment:
I'm a little confused about the logic here. What are you trying to do?

Reply:
So this is something called a bitmask, you can read an introduction to them [here](<https://bamblebam.medium.com/unmasking-bit-masks-71421d0eb2fa>) 🙂 

The `flags` variable will be some number representing a combination of flags (e.g. `0101`), and when we use a logic AND operator with a single flag (e.g. `0001`) we will get one of two results: either `0000` (zero) if our flag isn't present, or `0001` (our flag) if it is present.

In this case when we ask if `flags & NO_RIGHT_CHAINING == NO_RIGHT_CHAINING` we're basically asking whether the `NO_RIGHT_CHAINING` flag is present in our flags bitfield.

Reply:
I understand the usage of bitflags. I am confused about the way chaining is being checked here (the new flag's documentation unfortunately did not clarify it for me). Can you explain how the flags are intended to work? Or, more specifically, when these "maybe"/casting converters are intended to be allowed?